document/FileFormat.mdk

```diff
@@ -2857,9 +2865,28 @@ The value of `formatId` shall correspond to one of the names in the set of name/
 of the `rule` object (&sect;[#rule]) whose `id` property (&sect;[#rule-id]) matches the `ruleId` property (&sect;[#result-ruleId]) of this result.
 
 ### `arguments` property { #formattedMessage-arguments }
-A `formattedMessage` object may contain a property named `arguments`,
+If the message format string specified by `formatId` contains any placeholders,
+the `formattedMessage` object shall contain a property named `arguments`,
 whose value is an array of string values that will be used, in combination with a message format,
 to construct a result message.
+The array shall have as many elements are there are distinct placeholders in the message format.
```

Comment:
Yes, we do: "Each placeholder shall be of the form "`{`_n_`}`", where _n_ is a non-negative integer which represents a 0-based index into the list of arguments."

---
In reply to: [72723920](https://github.com/sarif-standard/sarif-spec/pull/255#discussion_r72723920) [](ancestors = 72723920)